### PR TITLE
docs: update the differential loading section

### PR DESCRIPTION
**Pull request by santoshyadavdev**
Angular 10 onward differential loading is not enabled by default, the text mentions it is the default for Angular CLI 8 and above which is irrelevant now.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by santoshyadavdev** [comment]
Hi Team,
Please let me know if any changes are required.

**Review by kapunahelewong — APPROVED** [review]
LGTM. Thank you, @santoshyadavdev! We'll just need engineering approval and we can move forward.

Reviewed-for: global-docs-approvers

**Comment by santoshyadavdev** [comment]
Hi @kapunahelewong, can we merge this, let me know If i need to do anything.

**Comment by santoshyadavdev** [comment]
Rebased and pushed again.

**Comment by santoshyadavdev** [comment]
Hi @gkalpak @aikidave can we review this PR please

File: aio/content/guide/deployment.md [hunk 1]
```diff
@@ -582,7 +582,7 @@ If your Browserslist configuration includes support for any legacy browsers, the
 
 ## Local development in older browsers
 
-In Angular CLI version 8 and higher, differential loading is enabled by default for the `ng build` command.
+In Angular CLI version 10 and above, differential loading is not enabled by default for the `ng build` command.
 The `ng serve`, `ng test`, and `ng e2e` commands, however, generate a single ES2015 build which cannot run in older browsers that don't support the modules, such as IE 11.
```
Comment (gkalpak) [inline aio/content/guide/deployment.md:586]: I feel this whole paragraph needs rewording. Saying that "differential loading is not enabled" and "the commands however generate a single build" sounds confusing.

Also, is differential loading really disabled? Or is it just that the browserslist configuration is such that there is no need for differential loading.

Maybe someone from the tooling team can help word this in an accurate way :smiley:

Comment (santoshyadavdev) [reply aio/content/guide/deployment.md:586]: Agreed, so the thing is this is due to the browserslist configuration, but only for new projects, but if you migrate it will generate both builds. I will try to rephrase it, may be @alan-agius4 can help.

File: aio/content/guide/deployment.md [hunk 2]
```diff
@@ -582,7 +582,7 @@ If your Browserslist configuration includes support for any legacy browsers, the
 
 ## Local development in older browsers
 
-In Angular CLI version 8 and higher, differential loading is enabled by default for the `ng build` command.
+In Angular CLI version 10 and above, differential loading is not enabled by default for the `ng build` command.
```
Comment (alan-agius4) [inline aio/content/guide/deployment.md:585]: Maybe?

```suggestion
Differential loading is not enabled by default for application projects that were generated with Angular CLI 10 and above.
```

Side node: this section does appear to be in the wrong part of the docs. Local development seems to be unrelated to deployment.

Comment (alan-agius4) [reply aio/content/guide/deployment.md:585]: Actually it appears that there are more unrelated section to deployment. I created a seperate issue https://github.com/angular/angular/issues/39735

Comment (santoshyadavdev) [reply aio/content/guide/deployment.md:585]: Thank alan, I will work on the issue you mentioned.

Comment (santoshyadavdev) [reply aio/content/guide/deployment.md]: So meanwhile the main issues get prioritized, I am updating the contents here, please check.

**Review by alan-agius4 — APPROVED** [review]
